lib/rubocop/cop/rspec/multiple_describes.rb

```diff
@@ -36,6 +101,59 @@ def on_top_level_group(node)
 
           add_offense(node.send_node)
         end
+
+        def on_send(node)
+          return unless style == :splitting
+
+          return unless describe?(node)
+
+          return if includes_common_codes?(node)
+
+          block_node = node.parent
+          return if root_node?(block_node)
+          return if includes_describe_in_previous_node?(block_node)
+
+          add_offense(node, message: MSG_SPLIT)
+        end
+
+        private
+
+        def includes_describe_in_previous_node?(block_node)
+          previous_block_node = previous_sibling(block_node)
+          return false if previous_block_node.nil?
+
+          if previous_block_node.child_nodes.any? { |child| describe?(child) }
+            return true
+          end
+
+          includes_describe_in_previous_node?(previous_block_node)
+        end
+
+        def includes_common_codes?(node)
+          root = root_node_for_current_node(node)
+
+          root.each_descendant(:block).any? do |child|
+            shared_example?(child) || custom_matcher?(child)
+          end
+        end
+
+        def previous_sibling(block_node)
+          return nil if (block_node.sibling_index - 1).negative?
+
+          block_node.parent.child_nodes[block_node.sibling_index - 1]
+        end
+
+        def root_node_for_current_node(node)
+          node.ancestors.find { |parent| root_node?(parent) }
```

Comment:
I believe this can be dramatically simplified to something like:
```ruby
node.ancestors(:begin).last
```

Reply:
I understand.
I think the suggestion code is nice, but because I wanted to be able to check the use of share groups or matcher, I changed the code.